### PR TITLE
Feature/migrate ci to new tenant

### PR DESCRIPTION
**Pull request by noraabiakar**
# PR for issue #1885 

## What is being addressed

Migrate the CI to the new Azure subscription (fdpo). In the process, switch to using Azure KeyVault for storing secrets instead of 
GitHub secrets. 

## How is this addressed

- Add a new GitHub action that, given a list of keys, will read the secrets from a KeyVault (in parallel) and expose the values as masked GitHub environment variables.
- Make the necessary changes to the e2e CI to work with the new way of reading secrets. 
- Address other small issues encountered while making the above changes.

### CODE REVIEW

File: .github/workflows/e2e-ci.yaml [hunk 1]
```diff
@@ -9,24 +9,27 @@ on: # rebuild any PRs and main branch changes
...
+        required: false
       TestsToRun:
         default: '[SensorDecodingTest,OTAAJoinTest,ABPTest,OTAATest,MacTest,ClassCTest,C2DMessageTest,MultiGatewayTest,MultiConcentratorTest,CupsTest,LnsDiscoveryTest,CloudDeploymentTest]'
         description: 'tests to run'
+        required: false
       TxPower:
         description: 'TXPower value to use in E2E tests'
         default: '6'
+        required: false
+permissions:
```
Comment (Mandur) [inline .github/workflows/e2e-ci.yaml:26]: Why is this needed?

Comment (noraabiakar) [reply .github/workflows/e2e-ci.yaml:26]: It is needed for OIDC authentication https://docs.github.com/en/actions/deployment/security-hardening-your-deployments/configuring-openid-connect-in-cloud-providers#adding-permissions-settings

**Review by Mandur — APPROVED** [review]
